File: platforms/hacker-rank/E--20--append-and-delete/solutions/solution.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'appendAndDelete' function below.
#
# The function is expected to return a STRING.
# The function accepts following parameters:
#  1. STRING s
#  2. STRING t
#  3. INTEGER k
#


def appendAndDelete(s, t, k):
    sn, tn = len(s), len(t)

    if k >= sn + tn:
        return "Yes"

    plen = 0

    for i in range(min(sn, tn)):
        if s[i] == t[i]:
            plen += 1
        else:
            break

    minops = sn + tn - 2 * plen
    if minops > k:
        return "No"

    if minops == k:
        return "Yes"

    if (minops - k) % 2 == 0:
        return "Yes"

    return "No"


if __name__ == "__main__":
    fptr = open(os.environ["OUTPUT_PATH"], "w")
    s = input()
    t = input()
    k = int(input().strip())
    result = appendAndDelete(s, t, k)
    fptr.write(result + "\n")
    fptr.close()

File: platforms/hacker-rank/E--20--append-and-delete/solutions/solution.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"strconv"
	"strings"
)

/*
 * Complete the 'appendAndDelete' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts following parameters:
 *  1. STRING s
 *  2. STRING t
 *  3. INTEGER k
 */

func appendAndDelete(s string, t string, k int32) string {
	sn, tn := len(s), len(t)
	if k >= int32(sn+tn) {
		return "Yes"
	}

	plen := 0
	for i := 0; i < min(sn, tn); i++ {
		if s[i] == t[i] {
			plen++
		} else {
			break
		}
	}

	minops := sn + tn - 2*plen
	if int32(minops) > k {
		return "No"
	} else if int32(minops) == k {
		return "Yes"
	} else {
		if (int32(minops)-k)%2 == 0 {
			return "Yes"
		} else {
			return "No"
		}
	}
}

func main() {
	reader := bufio.NewReaderSize(os.Stdin, 16*1024*1024)

	stdout, err := os.Create(os.Getenv("OUTPUT_PATH"))
	checkError(err)

	defer stdout.Close()

	writer := bufio.NewWriterSize(stdout, 16*1024*1024)

	s := readLine(reader)

	t := readLine(reader)

	kTemp, err := strconv.ParseInt(strings.TrimSpace(readLine(reader)), 10, 64)
	checkError(err)
	k := int32(kTemp)

	result := appendAndDelete(s, t, k)

	fmt.Fprintf(writer, "%s\n", result)

	writer.Flush()
}

func readLine(reader *bufio.Reader) string {
	str, _, err := reader.ReadLine()
	if err == io.EOF {
		return ""
	}

	return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
	if err != nil {
		panic(err)
	}
}
